src/Core/Scripts/Runtime/TypeSystem.js

```diff
@@ -11,13 +11,15 @@ function createType(typeName, typeInfo, typeRegistry) {
   // or a function, representing a record factory.
 
   if (Array.isArray(typeInfo)) {
-    var type = typeInfo[0];
-
+      var typeMarker = typeInfo[0];
+      var type = typeInfo[1];
+      var prototypeDescription = typeInfo[2];
+      var baseType = typeInfo[3];
+      var interfaces = typeInfo.slice(4);
     // A class is minimally the class type and an object representing
     // its prototype members, and optionally the base type, and references
     // to interfaces implemented by the class.
-    if (typeInfo.length >= 2) {
-      var baseType = typeInfo[2];
+    if (typeMarker == _classMarker) {
```

Comment:
Ideally we should be using the === Check here as the values should match in reference and value.